DataDefinitions/CommodityMarketQuote.cs

```diff
@@ -122,108 +122,141 @@ public CommodityMarketQuote(CommodityDefinition definition)
 
         public static CommodityMarketQuote FromCapiJson(JObject capiJSON)
         {
-            long eliteId = (long)capiJSON["id"];
-            string edName = (string)capiJSON["name"];
-            string category = (string)capiJSON["category"];
-            int meanPrice = (int)capiJSON["meanPrice"];
-
-            CommodityDefinition commodityDef = CommodityDefinition.CommodityDefinitionFromEliteID(eliteId) ?? CommodityDefinition.FromEDName(edName);
-            if (commodityDef == null || (string)capiJSON["name"] != commodityDef.edname)
+            CommodityMarketQuote quote = null;
+            try 
             {
-                if (commodityDef?.edname != "Drones")
+                long eliteId = (long)capiJSON["id"];
+                string edName = (string)capiJSON["name"];
+                string category = (string)capiJSON["category"];
+                int meanPrice = (int)capiJSON["meanPrice"];
+
+                // `CommodityDefinition.FromEDName()` would create a new commodity if we don't already know about the commodity.
+                // We'll defer using that, in favor of instead creating a more complete definition below.
+                CommodityDefinition commodityDef = CommodityDefinition.CommodityDefinitionFromEliteID(eliteId);
+                if (commodityDef is null)
                 {
-                    // Unknown commodity; report the full object so that we can update the definitions 
-                    Logging.Info("Commodity definition error: " + (string)capiJSON["name"], JsonConvert.SerializeObject(capiJSON));
+                    CommodityCategory commodityCategory = CommodityCategory.FromEDName(category);
+                    if (commodityCategory != CommodityCategory.NonMarketable)
```

Comment:
Just to be clear, we are now testing against `CommodityCategory.NonMarketable` rather than `commodityDef?.edname != "Drones"` but I think that is better.